File: Semana_2/Playwright/Java/maven/curso/e2e/src/test/java/com/curso/playwright/pages/separated/OpenAccountActions.java
package com.curso.playwright.pages.separated;

import com.curso.playwright.evidence.EvidenceSupport;
import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.microsoft.playwright.options.SelectOption;
import com.microsoft.playwright.options.WaitForSelectorState;

public class OpenAccountActions {
    private final Page page;
    private final OpenAccountLocators locators;

    public OpenAccountActions(Page page) {
        this.page = page;
        this.locators = new OpenAccountLocators(page);
    }

    public String openSavingsAccount() {
        locators.fromAccountOptions().first().waitFor(new Locator.WaitForOptions().setState(WaitForSelectorState.ATTACHED));
        locators.accountTypeSelect().selectOption(new SelectOption().setLabel("SAVINGS"));
        EvidenceSupport.capture(page, "parabank-separated-select-savings");
        locators.openAccountButton().click();
        EvidenceSupport.capture(page, "parabank-separated-click-open-account");
        locators.newAccountId().waitFor();
        String accountId = locators.newAccountId().innerText().trim();
        EvidenceSupport.capture(page, "parabank-separated-account-created");
        return accountId;
    }
}

File: Semana_2/Playwright/javascript/npm/curso/e2e/pages/separated/open-account.actions.js
const { OpenAccountLocators } = require('./open-account.locators');
const { capture } = require('../evidence');

class OpenAccountActions {
  constructor(page) {
    this.page = page;
    this.locators = new OpenAccountLocators(page);
  }

  async openSavingsAccount() {
    await this.locators.fromAccountOptions().first().waitFor({ state: 'attached' });
    await this.locators.accountTypeSelect().selectOption({ label: 'SAVINGS' });
    await capture(this.page, 'parabank-separated-select-savings');
    await this.locators.openAccountButton().click();
    await capture(this.page, 'parabank-separated-click-open-account');
    await this.locators.newAccountId().waitFor({ state: 'visible' });
    await capture(this.page, 'parabank-separated-account-created');
    return (await this.locators.newAccountId().innerText()).trim();
  }
}

module.exports = { OpenAccountActions };
